src/components/@shared/FormInput/InputElement/ConsumerParameters/DefaultInput.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { ReactElement } from 'react'
+import Input, { InputProps } from '../..'
+import { Field, useField } from 'formik'
+import { FormConsumerParameter } from '@components/Publish/_types'
+
+export default function DefaultInput({
+  index,
+  inputName,
+  ...props
+}: InputProps & {
+  index: number
+  inputName: string
+}): ReactElement {
+  const [field] = useField<FormConsumerParameter[]>(inputName)
+
+  const getStringOptions = (
+    options: { key: string; value: string }[]
+  ): string[] => {
+    if (!options?.length) return []
+
+    return options.map((option) => option.key)
+  }
+
+  return (
+    <Field
+      {...props}
+      component={Input}
+      type={
+        field.value[index].type === 'boolean'
+          ? 'select'
+          : field.value[index].type
+      }
+      options={
+        field.value[index].type === 'boolean'
+          ? ['true', 'false']
+          : field.value[index].type === 'select'
+          ? getStringOptions(field.value[index]?.options)
+          : field.value[index].options
+      }
+    />
+  )
```

Comment:
suggestion(non-blocking): We are accessing `field.value[index].type` and `field.value[index].options` multiple times here. You could store this value in a variable to avoid repeating yourself.

```
const fieldType = field.value[index].type;
const fieldOptions = field.value[index].options;
```